### ISSUE TITLE
English experiments

### ISSUE THREAD

**Issue by mikeizbicki** [n8stringham/gensim-evaluations#1]
The first task is to take your code and apply it to Mikolov's original dataset.  

We want to generate a graph that has three lines: one for each the analogy performance, and one for each of your methods (possibly more lines for variations on your methods).

The x-axis should be the fraction of the original training data used (probably log scaled).

**Comment by n8stringham** [comment]
Figured out how to load 20-newsgroups dataset from gensim-data. Was able to get a model preprocessed and trained.

 Next step is to run `evaluate_word_analogies` on these word vectors using Mikolov analogy test set. Will also need to work on adapting that test set to work with my other evaluation methods.

**Comment by n8stringham** [comment]
Today I spent a good chunk of time downloading the larger wikipedia dataset available from gensim and using it to train an out of the box word2vec model. While the dataset was downloading I worked on adapting the mikolov analogy set for use with my evaluation methods. This involved extracting all of the words from the analogy set and grouping them by category. Just waiting on the model to finish training so that I can run some evaluations.

**Comment by n8stringham** [comment]
Forgot to update from friday...Adjusted training script to train models on proportions of the wikipedia dataset. Proportions are currently based off of total words, but also tracking unique words so that we can train models for that metric as well.

**Comment by n8stringham** [comment]
Fixed the issue with the training loop and set it up to run as a shell script. Something to do with how I was handling the data streaming. Seems to be working now. Have models trained for sizes 2^10 up to 2^30. Next going to start running evaluations for those models so that I can generate a figure.

**Comment by n8stringham** [comment]
Worked on adding the randomization to data streaming so that instead of processing data in the order it is streamed, we randomly choose which sections to process. I created a [linear congruential generator](https://en.wikipedia.org/wiki/Linear_congruential_generator) which randomly assigns a number to each section, but have some questions regarding implementation to talk to you about tomorrow.

**Comment by n8stringham** [comment]
finished training models with randomized training data. Vocab sizes range from 54 words up to ~2.2million. Started running some evaluations and produced a simple plot.

**Comment by n8stringham** [comment]
Wrote up some pseudocode for odd-one-out and topk similarity evaluation algorithms. Read the paper for this weeks discussion. Started running evaluations on full test set ~24hrs ago....it's still going. Hopefully it's finished by tomorrow am
